lib/schema.js

```diff
@@ -206,7 +216,12 @@ const createInstance = (
   const domain = domains.get(name);
   const type = typeof value;
   const expected = domain ? domain.type : name;
-  if (type === expected) return value;
+  const domainType = domain ? domain.constructor.name : null;
+  const isValueDecorator = domainType in valueDecoratorValidators;
+  const validate = valueDecoratorValidators[domainType];
+  if (type === expected || isValueDecorator && validate(domain, value)) {
+    return value;
```

Comment:
Double check this returns scalar value, because `createInstance` intended to duplicate category calling factory and pass  copies of scalar values because scalars are passing by value. Do we pass flags and enums by value?

Reply:
Not at all, we pass flags as array of values

Reply:
array will be passed by reference